src/task01.java

```diff
@@ -0,0 +1,93 @@
+
+interface control {
+
+    void speed();
+}
+
+abstract class Vehicle implements control{
+
+    String type;
+    public Vehicle(String typ)
+    {
+        this.type = typ;
```

Comment:
we use `this` keyword when the passed arguments is same as the class's data member. This can be replaced with `type = typ` simply/